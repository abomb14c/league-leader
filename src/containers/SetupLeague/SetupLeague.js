import React, { Component } from 'react'; 

export class SetupLeauge extends Component {
  constructor(props){
    super(props)

    this.state = {

    }
  }

  render() {
    return (
      <div>
        create
      </div>
    )
  }
}

export default SetupLeague; 