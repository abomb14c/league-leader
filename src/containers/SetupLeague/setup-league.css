.league-menu {
  border: none;
  background-color: #C63D0F;
  color: #FDF3E7;
  height: 35px;
  font-size: 14px;
  margin-bottom: 4%;
  width: 52%;
}

.league-menu:hover {
  border: 1px solid #C63D0F;
  background: transparent;
}

.menu {
  width: 100%;
}

.setup-container{
  align-items: center;
  background-color: #3B3738;
  box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.30);
  display: flex;
  flex-direction: column;
  height: 470px;
  margin-top: 2%;
  width: 35%;
}

.setup-english {
  border: 1px solid #7E8F7C;
  background: transparent;
  color: #FDF3E7;
  margin-right: 2%;
  height: 30px;
  width: 30%;
}

.setup-english:hover, .setup-nba:hover {
  background: #7E8F7C;
}

.setup-input {
  height: 30px;
  font-size: 14px;
  margin-bottom: 7%;
  padding-left: 5px;
  width: 50%;
}

.setup-nba {
  border: 1px solid #7E8F7C;
  background: transparent;
  color: #FDF3E7;
  margin-right: 2%;
  height: 30px;
  width: 10%;
}

.setup-section {
  align-items: center;
  display: flex;
  justify-content: center;
}

.submit-setup {
  position: relative;
  top: 80px;
  border: none;
  background-color: #7E8F7C;
  color: #FDF3E7;
  height: 35px;
  font-size: 14px;
  margin-bottom: 4%;
  width: 52%;
}

.submit-setup:disabled {
  background-color: #636e72;
}

.setup-title {
  color: #FDF3E7;
  font-size: 28px;
  margin-bottom: 10%;
}