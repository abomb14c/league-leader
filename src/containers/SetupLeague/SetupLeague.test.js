import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import  { SetupLeague, mapDispatchToProps}  from './SetupLeague';

describe('LoginUser', () => {
  let wrapper;
  let mockCreateNewLeague;
  let mockHandleDraftTeams;
  let mockProps;

  beforeEach(() => {
    mockCreateNewLeague = jest.fn();
    mockHandleDraftTeams = jest.fn();
    mockProps = {
      EPL: [{}, {}, {}],
      NBA: {'EASTERN CONFERENCE':[{}, {}, {}], 
        'WESTERN CONFERENCE': [{}, {}, {}]}
    };

    wrapper = shallow(<SetupLeague 
      handleDraftTeams={mockHandleDraftTeams}
      createNewLeague={mockCreateNewLeague}
      {...mockProps}
    />);
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


  it("has a default state", () => {
    let expected = {
      showMenu: false,
      leagueName: '',
      leagueBet: '',
      league: ''
    };

    expect(wrapper.state()).toEqual(expected);
  });
});