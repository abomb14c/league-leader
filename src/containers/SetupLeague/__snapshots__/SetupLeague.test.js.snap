// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginUser should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SetupLeague
    EPL={
        Array [
            Object {},
            Object {},
            Object {},
          ]
    }
    NBA={
        Object {
            "EASTERN CONFERENCE": Array [
              Object {},
              Object {},
              Object {},
            ],
            "WESTERN CONFERENCE": Array [
              Object {},
              Object {},
              Object {},
            ],
          }
    }
    createNewLeague={[Function]}
    handleDraftTeams={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="setup-container"
>
        <h3
                className="setup-title"
        >
                Setup Your League
        </h3>
        <input
                className="setup-input"
                name="leagueName"
                onChange={[Function]}
                placeholder="League Name"
                value=""
        />
        <input
                className="setup-input"
                name="leagueBet"
                onChange={[Function]}
                placeholder="Place Your Bet"
                value=""
        />
        <button
                className="league-menu"
                onClick={[Function]}
        >
                Pick A League
        </button>
        <button
                className="submit-setup"
                disabled={true}
                onClick={[Function]}
        >
                Create League
        </button>
</div>,
      "className": "setup-section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="setup-title"
>
            Setup Your League
</h3>,
          <input
            className="setup-input"
            name="leagueName"
            onChange={[Function]}
            placeholder="League Name"
            value=""
/>,
          <input
            className="setup-input"
            name="leagueBet"
            onChange={[Function]}
            placeholder="Place Your Bet"
            value=""
/>,
          <button
            className="league-menu"
            onClick={[Function]}
>
            Pick A League
</button>,
          null,
          <button
            className="submit-setup"
            disabled={true}
            onClick={[Function]}
>
            Create League
</button>,
        ],
        "className": "setup-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Setup Your League",
            "className": "setup-title",
          },
          "ref": null,
          "rendered": "Setup Your League",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "setup-input",
            "name": "leagueName",
            "onChange": [Function],
            "placeholder": "League Name",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "setup-input",
            "name": "leagueBet",
            "onChange": [Function],
            "placeholder": "Place Your Bet",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pick A League",
            "className": "league-menu",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Pick A League",
          "type": "button",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create League",
            "className": "submit-setup",
            "disabled": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Create League",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="setup-container"
>
          <h3
                    className="setup-title"
          >
                    Setup Your League
          </h3>
          <input
                    className="setup-input"
                    name="leagueName"
                    onChange={[Function]}
                    placeholder="League Name"
                    value=""
          />
          <input
                    className="setup-input"
                    name="leagueBet"
                    onChange={[Function]}
                    placeholder="Place Your Bet"
                    value=""
          />
          <button
                    className="league-menu"
                    onClick={[Function]}
          >
                    Pick A League
          </button>
          <button
                    className="submit-setup"
                    disabled={true}
                    onClick={[Function]}
          >
                    Create League
          </button>
</div>,
        "className": "setup-section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="setup-title"
>
              Setup Your League
</h3>,
            <input
              className="setup-input"
              name="leagueName"
              onChange={[Function]}
              placeholder="League Name"
              value=""
/>,
            <input
              className="setup-input"
              name="leagueBet"
              onChange={[Function]}
              placeholder="Place Your Bet"
              value=""
/>,
            <button
              className="league-menu"
              onClick={[Function]}
>
              Pick A League
</button>,
            null,
            <button
              className="submit-setup"
              disabled={true}
              onClick={[Function]}
>
              Create League
</button>,
          ],
          "className": "setup-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Setup Your League",
              "className": "setup-title",
            },
            "ref": null,
            "rendered": "Setup Your League",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "setup-input",
              "name": "leagueName",
              "onChange": [Function],
              "placeholder": "League Name",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "setup-input",
              "name": "leagueBet",
              "onChange": [Function],
              "placeholder": "Place Your Bet",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Pick A League",
              "className": "league-menu",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Pick A League",
            "type": "button",
          },
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create League",
              "className": "submit-setup",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Create League",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
